### PR TITLE
fix: throw exception-like object to sentry

### PR DESCRIPTION
**Pull request by ohansFavour**
## Cause of issue

The error passed to `Sentry.captureException ` was not an exception-like object.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/25397(https://github.com/appsmithorg/appsmith/issues/25397)

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/sagas/ErrorSagas.tsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ export function* validateResponse(
   yield put({
     type: ReduxActionErrorTypes.API_ERROR,
     payload: {
-      error: response.responseMeta.error,
+      error: new Error(response.responseMeta.error.message),
```
Comment (Rishabh-Rathod) [inline app/client/src/sagas/ErrorSagas.tsx:121]: @ohansFavour Could you please add a description explaining why it causes a sentry exception? 

I noticed that the `payload.error` is read in code using `get(error, "message")` which should have worked fine.

**Review by Rishabh-Rathod — APPROVED** [review]

File: app/client/src/sagas/ErrorSagas.tsx (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/sagas/ErrorSagas.tsx:121]: @Rishabh-Rathod that is not where the problem is. The problem occurs in Ln 247

`yield call(Sentry.captureException, error)`. The error being sent to `Sentry.captureException` is not an exception-type object. The error message below depicts that.

Screenshot 2023-07-19 at 10 01 49

Comment (ohansFavour) [reply app/client/src/sagas/ErrorSagas.tsx:121]: Let me know what you think, before I merge this.

Comment (Rishabh-Rathod) [reply app/client/src/sagas/ErrorSagas.tsx:121]: `apiFailureResponseInterceptor` has `Sentry.captureException(error);` where the error is not javascript Error object. According to the [docs](https://docs.sentry.io/platforms/javascript/usage/?original_referrer=https%3A%2F%2Fwww.google.com%2F#capturing-errors), it seems like the captureException method expects an Error object.

**Pull request merged**